08-read-write-plot.qmd

```diff
@@ -238,7 +239,46 @@ gpd.read_file(u, layer='Placemarks')
 
 ### Raster data
 
-...
+Similar to vector data, raster data comes in many file formats with some of them supporting multilayer files. 
+`rasterio.open` is used to create a file connection to a raster file, which can be subsequently used to read the metadata and/or the values, as shown previously (@sec-using-rasterio). 
+For example: 
+
+```{python}
+src = rasterio.open('data/srtm.tif')
+src
+```
+
+All of the previous examples read spatial information from files stored on your hard drive. 
+However, GDAL also allows reading data directly from online resources, such as HTTP/HTTPS/FTP web resources. 
+The only thing we need to do is to add a `/vsicurl/` prefix before the path to the file. 
+Let's try it by connecting to the global monthly snow probability at 500 m resolution for the period 2000-2012 (T. Hengl 2021 add reference...). 
+Snow probability for December is stored as a Cloud Optimized GeoTIFF (COG) file (see @sec-file-formats). 
+To read an online file, we just need to provide its URL together with the `/vsicurl/` prefix:
+
+```{python}
+url = "/vsicurl/https://zenodo.org/record/5774954/files/clm_snow.prob_esacci.dec_p.90_500m_s0..0cm_2000..2012_v2.0.tif"
```

Comment:
Love the use of Zenodo. Worth adding a hyperlink to the underlying research? Hopefully the URL fits on a single line, if not no big deal... We can cross that bridge when we get closer to publication of the physical book!

Reply:
Sure, that's a good idea!